modules/ensemble/lib/framework/device.dart

```diff
@@ -13,22 +14,95 @@ import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
 import 'package:get_it/get_it.dart';
 import 'package:intl/intl.dart';
 
-/// get device information as well as requesting device permissions
-class Device
-    with
-        Invokable,
-        MediaQueryCapability,
-        LocationCapability,
-        DeviceInfoCapability {
+class Device with Invokable, MediaQueryCapability, LocationCapability, DeviceInfoCapability, WidgetsBindingObserver {
   static final Device _instance = Device._internal();
+  static bool _isInitialized = false;
+  bool _isHandlingChange = false;
+  MediaQueryData? _lastReportedData;
+
+  static final StreamController<MediaQueryData> _deviceUpdateController = 
+      StreamController<MediaQueryData>.broadcast();
+  static Stream<MediaQueryData> get onDeviceUpdate => _deviceUpdateController.stream;
+  
+  Device._internal() {
+    if (!_isInitialized) {
+      print('🔄 Initializing Device observer...');
+      WidgetsBinding.instance.addObserver(this);
+      // Set up initial data
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        _handleMediaQueryChange();  // Call this to set initial state
+      });
+      _isInitialized = true;
+    }
+  }
+
+  factory Device() => _instance;
+
+  @override
+  void didChangeMetrics() {
+    print('📏 Metrics changed - handling update...');
+    if (!_isHandlingChange) {
+      WidgetsBinding.instance.addPostFrameCallback((_) => _handleMediaQueryChange());
+    }
+  }
+
+  void _handleMediaQueryChange() {
+    if (_isHandlingChange) return;
+    _isHandlingChange = true;
+    
+    try {
+      final context = Utils.globalAppKey.currentContext;
+      if (context == null) {
+        print('❌ No context available for MediaQuery update');
+        return;
+      }
+
+      final newData = MediaQuery.of(context);
+      
+      // Check for real changes
+      if (_hasSignificantChanges(_lastReportedData, newData)) {
+        
+        _lastReportedData = newData;
+        MediaQueryCapability.data = newData;
+        
+        // Broadcast the change
+        print('📢 Broadcasting device update...');
+        _deviceUpdateController.add(newData);
+      }
+    } catch (e) {
+      print('💥 Error handling MediaQuery change: $e');
+    } finally {
+      _isHandlingChange = false;
+    }
+  }
+  
+  bool _hasSignificantChanges(MediaQueryData? oldData, MediaQueryData newData) {
```

Comment:
Maybe you use this for testing but why not just remove this function and change it to "orienttation != newData.orientation || size != newData.size". Simpler and less code to look through